projects/transformers/run.py

```diff
@@ -334,9 +354,42 @@ def run_finetuning_single_task(
     if training_args.do_train:
         train(trainer, training_args.output_dir, last_checkpoint)
 
+    # In order to get time series of evaluation metrics, you need to grab the
+    # TrackEvalMetrics callback instance. This means you need to know its index
+    # in the model_args.trainer_callbacks list. This block finds it.
+    tracked_metrics = False
+    tracked_metrics_idx = None
+    for callback_idx in range(len(model_args.trainer_callbacks)):
+        if isinstance(model_args.trainer_callbacks[callback_idx], TrackEvalMetrics):
+            tracked_metrics = True
+            tracked_metrics_idx = callback_idx
+
+    # If already tracking metrics, no need to evaluate again at the very end
```

Comment:
Yes, good catch. If using load_best_model_at_end though, it would be redundant. I'll update this block to handle these cases properly, and also move it to a separate function as Lucas suggested.